.pre-commit-config.yaml

```diff
@@ -44,6 +44,11 @@ repos:
             # Expected files should be left verbatim.
             \.expected(-failure)?$
           )
+- repo: https://github.com/arenadotio/pre-commit-ocamlformat
```

Comment:
Isn't this going to apply ocamlformat to the whole repo?

Reply:
Ok, I missed the -ignore files because I skipped the `[[fmt] run dune fmt](https://github.com/herd/herdtools7/pull/785/commits/29eaed86c6cb0363ea6e0fb506edf4c785461154)` because I thought it was just checking in the formatting changes but it does add some configuration as well. If we want to retain the commits in this PR can we add these changes as a separate commit?